### ISSUE TITLE
https POST request has blocked by CORS policy 

### ISSUE THREAD

**Issue by raishvaria** [nelmio/NelmioCorsBundle#159]
Hi,

I have an issue with https POST request using this configuarathis nelmio_cor yaml configuaration
```nelmio_cors:
    defaults:
        origin_regex: true
        allow_methods: [ 'GET', 'OPTIONS', 'POST', 'PUT', 'PATCH', 'DELETE' ]
        allow_headers: [ 'Content-Type', 'Authorization' ]
        expose_headers: ['Link']
        max_age: 3600
    paths:  
        '^/api/':
            allow_methods: ['POST', 'PUT', 'GET', 'DELETE']
            max_age: 3600
````

![Screenshot 2020-12-19 at 5 14 58 PM](https://user-images.githubusercontent.com/26214655/102688769-e7c79180-421e-11eb-8397-a98e5e552241.png)

Also it's working on GET request using same nelmio configuration. But i'm trying to add allow origin `*` on this nelmeio yaml the POST request working find but GET Request got multiple header issue.

```nelmio_cors:
    defaults:
        origin_regex: true
        allow_origin: ['%env(CORS_ALLOW_ORIGIN)%']
        allow_methods: [ 'GET', 'OPTIONS', 'POST', 'PUT', 'PATCH', 'DELETE' ]
        allow_headers: [ 'Content-Type', 'Authorization' ]
        expose_headers: ['Link']
        max_age: 3600
    paths:  
        '^/api/':
            allow_methods: ['POST', 'PUT', 'GET', 'DELETE']
            max_age: 3600
```
![Screenshot 2020-12-19 at 5 30 20 PM](https://user-images.githubusercontent.com/26214655/102688898-1003c000-4220-11eb-8648-435a8bc14288.png)
Any solution? 
Thanks in advance.

**Comment by fd6130** [comment]
Are you sure your address `localhost:8899` is allowed to be access by any address? `localhost:3000` and `localhost:8899` are different though.

**Comment by raishvaria** [comment]
Hi @fd6130   
My frontend host is `localhost:3000` and backend symfony running on docker host is `localhost:8899`.

**Comment by fd6130** [comment]
> Hi @fd6130 Yes it's work when using the GET request from `localhost:8899` but POST is not working.
> 
> My frontend host is `localhost:3000` and backend symfony running on docker host is `localhost:8899`.

Can you check if the POST request is working without CORS setup? Sometime CORS occurs because the code got problem.

**Comment by raishvaria** [comment]
> > Hi @fd6130 Yes it's work when using the GET request from `localhost:8899` but POST is not working.
> > My frontend host is `localhost:3000` and backend symfony running on docker host is `localhost:8899`.
> 
> Can you check if the POST request is working without CORS setup? Sometime CORS occurs because the code got problem.

No it's not working without CORS setup

**Comment by fd6130** [comment]
> > > Hi @fd6130 Yes it's work when using the GET request from `localhost:8899` but POST is not working.
> > > My frontend host is `localhost:3000` and backend symfony running on docker host is `localhost:8899`.
> > 
> > 
> > Can you check if the POST request is working without CORS setup? Sometime CORS occurs because the code got problem.
> 
> No it's not working without CORS setup

What kind of error you get from the POST request without CORS? And without CORS you should be able to access any resources from your backend server.

**Comment by fd6130** [comment]
> > > Hi @fd6130 Yes it's work when using the GET request from `localhost:8899` but POST is not working.
> > > My frontend host is `localhost:3000` and backend symfony running on docker host is `localhost:8899`.
> > 
> > 
> > Can you check if the POST request is working without CORS setup? Sometime CORS occurs because the code got problem.
> 
> No it's not working without CORS setup

Are you using Symfony 5?

**Comment by fd6130** [comment]
How about change your path to this one  (simply copy paste from default actually) ?

```
paths:  
        '^/api/':
            origin_regex: true
            allow_origin: ['*']
            allow_methods: [ 'GET', 'POST', 'PUT', 'DELETE' ]
            allow_headers: [ 'Content-Type', 'Authorization' ]
            expose_headers: ['Link']
            max_age: 3600
```

**Comment by dunglas** [comment]
It's because you used `*` as origin. Explicitly use the name of the domain instead, (or a regex if you want to match several domains). See https://stackoverflow.com/a/19744754/1352334 for some explanations.

Duplicate of https://github.com/api-platform/core/issues/3601.

**Comment by djpremier** [comment]
Sorry to reopen "Dunglas", but this error is different. The CORS with * occurs due to access denial, this refers to the possibility of making the PUT / PATCH request for the API, the other methods work correctly. Even in the old code (which I built directly in Symfony with the integration) it works with the same code. So in my opinion it was a change or something that is done differently from the docker code, in relation to the integrated in Symfony.

And in my case the regex is this:
`CORS_ALLOW_ORIGIN=^https?://(localhost|127\.0\.0\.1|MY_ADDRESS)(:[0-9]+)?$`
